### ISSUE TITLE
Allow `unknown` to be thrown (by default or through some configuration)

### ISSUE THREAD

**Issue by guilhermesimoes** [cartant/eslint-plugin-etc#39]
Using both these rules:
```ts
'etc/no-implicit-any-catch': 'error',
'etc/throw-error': 'error',
```
makes the following code always throw an error:
```ts
.catch((error: unknown) => {
  Logger.error(error);
  throw error;
});
```
It would be great if it was possible to rethrow an unknown error.

**Comment by cartant** [comment]
Should be fixed in 2.0.1.
